src/client/AppEmbeddedSDK.ts

```diff
@@ -0,0 +1,63 @@
+import { randomString } from './utils';
+
+const ACTION_ID_LENFTH = 80;
+
+export class AppEmbeddedSDK {
+    private listener: (this: Window, ev: MessageEvent) => any;
+    private callbacks: Map<string, (response: any) => any>;
+
+    constructor() {
+        this.listener = () => console.log('init');
+        this.callbacks = new Map();
+    }
+
+    public getUserInfo(): Promise<any> {
+        return this.call(AppEmbeddedSDKActions.GET_USER_INFO);
+    }
+
+    public getRoomInfo(): Promise<any> {
+        return this.call(AppEmbeddedSDKActions.GET_ROOM_INFO);
+    }
+
+    /**
+     * Create a message listener with id and add them to callbacks map set.
+     * @param id the id of the message event listener
+     */
+    private initListener(): void {
```

Comment:
I guess this method could be public, so that the developer of the `externalComponent` can choose when to initialise the Apps-Engine UI lib. Then you can remove the `this.initListener` call from the `call` method

Reply:
OK. I try it.

Reply:
Thanks for your suggestion! I've make the initListener to be public.